wgpu-hal/src/lib.rs

```diff
@@ -522,6 +535,61 @@ pub trait Adapter: WasmNotSendSync {
     unsafe fn get_presentation_timestamp(&self) -> wgt::PresentationTimestamp;
 }
 
+/// A connection to a GPU and a pool of resources to use with it.
+///
+/// A `wgpu-hal` `Device` represents an open connection to a specific graphics
+/// processor, controlled via the backend [`Device::A`]. This corresponds to a
+/// backend object like a Vulkan logical device ([`vkDevice`]) or a Direct3D 12
+/// virtual adapter ([`ID3D12Device`]).
```

Comment:
If mentioning D3D12 stuff is confusing, I'll just leave it out.

Reply:
Actually, since `wgpu_hal::dx12::Device` is definitely an `ID3D12Device` with metadata, I think this paragraph is true as written. The intent here is to give people familiar with one or the other platform API an intuition as to what one does with a `wgpu_hal::Device`, and `ID3D12Device` is the right analog.

So I'm going to leave this as is, and put a comment on `dx12::Adapter::device` to head off confusion.

Reply:
On chat, cwfitzgerald explained to me that `ID3D12Device`s are singletons per adapter. I also looked through the Metal backend a bit. I think I've got some new text that covers the situation well enough.